oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/ServerApi.java

```diff
@@ -0,0 +1,567 @@
+
+    @Named("servers:privatenetwork:list")
+    @GET
+    @Path("/{serverId}/private_networks")
+    @ResponseParser(ServerApi.PrivateNetworkListParser.class)
+    @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+    List<ServerPrivateNetwork> listPrivateNetworks(@PathParam("serverId") String serverId);
+
+    @Named("servers:privatenetwork:create")
+    @POST
+    @Path("/{serverId}/private_networks")
+    @MapBinder(BindToJsonPayload.class)
+    @ResponseParser(ServerApi.ServerParser.class)
+    Server assignPrivateNetwork(@PathParam("serverId") String serverId, @PayloadParam("id") String privateNetworkId);
+
+    @Named("servers:privatenetwork:get")
+    @GET
+    @Path("/{serverId}/private_networks/{privateNetworkId}")
+    @ResponseParser(ServerApi.PrivateNetworkParser.class)
+    @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+    ServerPrivateNetwork getPrivateNetwork(@PathParam("serverId") String serverId, @PathParam("privateNetworkId") String privateNetworkId);
+
+    @Named("servers:privatenetwork:delete")
+    @DELETE
+    @Path("/{serverId}/private_networks/{privateNetworkId}")
+    @ResponseParser(ServerApi.ServerParser.class)
+    @MapBinder(BindToJsonPayload.class)
+    @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+    Server deletePrivateNetwork(@PathParam("serverId") String serverId, @PathParam("privateNetworkId") String privateNetworkId);
+
+    @Named("servers:snapshot:list")
+    @GET
+    @Path("/{serverId}/snapshots")
+    @ResponseParser(ServerApi.SnapshotListParser.class)
+    @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+    List<Snapshot> listSnapshots(@PathParam("serverId") String serverId);
+
+    @Named("servers:snapshot:create")
+    @POST
+    @Path("/{serverId}/snapshots")
+    @Produces("application/json")
+    @ResponseParser(ServerApi.ServerParser.class)
+    Server createSnapshot(@PathParam("serverId") String serverId);
+
+    @Named("servers:snapshot:update")
+    @PUT
+    @Path("/{serverId}/snapshots/{snapshotId}")
+    @ResponseParser(ServerApi.ServerParser.class)
+    @MapBinder(BindToJsonPayload.class)
+    @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+    Server restoreSnapshot(@PathParam("serverId") String serverId, @PathParam("snapshotId") String snapshotId);
+
+    @Named("servers:snapshot:delete")
+    @DELETE
+    @Path("/{serverId}/snapshots/{snapshotId}")
+    @ResponseParser(ServerApi.ServerParser.class)
+    @MapBinder(BindToJsonPayload.class)
+    @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+    Server deleteSnapshot(@PathParam("serverId") String serverId, @PathParam("snapshotId") String snapshotId);
+
+    @Named("servers:clone:create")
+    @POST
+    @Path("/{serverId}/clone")
+    @ResponseParser(ServerApi.ServerParser.class)
+    Server clone(@PathParam("serverId") String serverId, @BinderParam(BindToJsonPayload.class) Server.Clone clone);
+
+    static final class ServerListParser extends ParseJson<List<Server>> {
```

Comment:
jclouds should be able to deserialize these objects out of the box. Can all the parsers in this class be removed, and the corresponding response parser annotations in the api methods?